pkg/validator/validators.go

```diff
@@ -266,3 +268,34 @@ func (kc *policyValidator) upstreamTrafficSettingValidator(req *admissionv1.Admi
 
 	return nil, nil
 }
+
+func meshRootCertificateValidator(req *admissionv1.AdmissionRequest) (*admissionv1.AdmissionResponse, error) {
+	switch req.Operation {
+	case admissionv1.Create:
+		newMRC := &configv1alpha2.MeshRootCertificate{}
+		if err := json.NewDecoder(bytes.NewBuffer(req.Object.Raw)).Decode(newMRC); err != nil {
+			return nil, err
+		}
+
+		if newMRC.Spec.TrustDomain == "" {
```

Comment:
nit: add a new func validateMRC() error

We will likely want to add new validations, and it can be reused in the update as well.

Reply:
do we also need to check that `Spec.Provider` information is set here?

Reply:
Good call out. I could add validation for the required string value fields

Reply:
Added `Spec.Provider` validation

Reply:
And more unit tests :)